src/Confluent.Kafka/Internal/Util.cs

```diff
@@ -28,19 +29,33 @@ internal static class Marshal
             /// <summary>
             ///     Interpret a zero terminated c string as UTF-8.
             /// </summary>
-            public static string PtrToStringUTF8(IntPtr strPtr)
+            public unsafe static string PtrToStringUTF8(IntPtr strPtr)
             {
                 // TODO: Is there a built in / vectorized / better way to implement this?
                 var length = 0;
-                unsafe
-                {
-                    byte* pTraverse = (byte*)strPtr;
-                    while (*pTraverse != 0) { pTraverse += 1; }
-                    length = (int)(pTraverse - (byte*)strPtr);
-                }
+                
+                byte* pTraverse = (byte*)strPtr;
+                while (*pTraverse != 0) { pTraverse += 1; }
+                length = (int)(pTraverse - (byte*)strPtr);
+#if NET45
                 var strBuffer = new byte[length];
-                System.Runtime.InteropServices.Marshal.Copy(strPtr, strBuffer, 0, length);
+                System.Runtime.InteropServices.Marshal.Copy(strPtr, strBuffer, 0, strBuffer.Length);
```

Comment:
the compiler probably optimizes this away, but isn't this exactly the same as the original (but possibly less efficient?)

Reply:
agree. pushed the fix